source/fuzz/fuzzer_pass_add_parameters.cpp

```diff
@@ -57,50 +57,11 @@ void FuzzerPassAddParameters::Apply() {
       continue;
     }
 
-    const auto* type_inst =
-        fuzzerutil::GetFunctionType(GetIRContext(), &function);
-    assert(type_inst);
-
-    // -1 because we don't take return type into account.
-    auto num_old_parameters = type_inst->NumInOperands() - 1;
-    auto num_new_parameters =
-        GetFuzzerContext()->GetRandomNumberOfNewParameters(
-            GetNumberOfParameters(function));
-
-    std::vector<uint32_t> all_types(num_old_parameters);
-    std::vector<uint32_t> new_types(num_new_parameters);
-    std::vector<uint32_t> parameter_ids(num_new_parameters);
-    std::vector<uint32_t> constant_ids(num_new_parameters);
-
-    // Get type ids for old parameters.
-    for (uint32_t i = 0; i < num_old_parameters; ++i) {
-      // +1 since we don't take return type into account.
-      all_types[i] = type_inst->GetSingleWordInOperand(i + 1);
-    }
-
-    for (uint32_t i = 0; i < num_new_parameters; ++i) {
-      // Get type ids for new parameters.
-      new_types[i] =
-          type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)];
-
-      // Create constants to initialize new parameters from.
-      constant_ids[i] = FindOrCreateZeroConstant(new_types[i]);
-    }
-
-    // Append new parameters to the old ones.
-    all_types.insert(all_types.end(), new_types.begin(), new_types.end());
-
-    // Generate result ids for new parameters.
-    for (auto& id : parameter_ids) {
-      id = GetFuzzerContext()->GetFreshId();
-    }
-
-    auto result_type_id = type_inst->GetSingleWordInOperand(0);
-    ApplyTransformation(TransformationAddParameters(
-        function.result_id(),
-        FindOrCreateFunctionType(result_type_id, all_types),
-        std::move(new_types), std::move(parameter_ids),
-        std::move(constant_ids)));
+    ApplyTransformation(TransformationAddParameter(
+        function.result_id(), GetFuzzerContext()->GetFreshId(),
+        FindOrCreateZeroConstant(
+            type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)]),
+        GetFuzzerContext()->GetFreshId()));
```

Comment:
Ah - I see that you've changed the fuzzer pass to only add a single parameter.

What I had meant was that the *transformation* should just add one parameter, but that the fuzzer pass would still add potentially many parameters by applying multiple transformations.

Can you change the fuzzer pass back so that it still queries the fuzzer context to decide how many parameters to add, and then loops N times, each time deciding on the type of the parameter and applying a transformation?

Reply:
Fixed in 858e2ce.